source/develop/release-management/features/sla/hosted-engine-edit-configuration-on-shared-storage.html.md

```diff
@@ -0,0 +1,137 @@
+---
+title: hosted-engine-edit-configuration-on-shared-storage
+category: feature
+authors: jtokar
+feature_name: Allow updating Hosted Engine configuration on shared storage
+feature_modules: sla
+feature_status: Design / partly implementaion stage
+---
+
+# Allow updating Hosted Engine configuration on shared storage
+
+## Summary
+Currently the hosted engine configuration is saved in a tar archive on a shared storage without any way of editing or updating it. This will allow users to easily update and get the shared configuration. 
+
+Currently there are two configuration archives on the shared storage, one contains the vm ovf and the other contains the hosted engine configuration.
+For the first stage this feature will provide the ability to edit broker.conf (that is located inside the configuration archive), for the second stage support for editing parts of the vm ovf will be added.
+
+## Detailed Description
+To achieve this new functionalities will be added to the hosted engine client:
+ 1. Set a new value
+ 2. Get the current value for a specific key
+ 3. Get a list of all keys and their current value
+ 4. Download a configuration archive
+ 5. Upload a configuration archive
+
+### Handling keys collisions:
+Since there are two (and some day maybe more) editable files, the get and set methods will receive an optional argument: “type”.
+This argument can be one of two options:
+1.  broker
+2.  vm
+
+In case of a collision the methods will return an error message with a request to use the desired type: “Duplicate key, please specify the key type.”
+
+
+### Handling synchronization on the shared storage - optimistic locking:
+To avoid sync issues, before updating the archive, an md5 checksum will be calculated. A new archive will be created, and after that if the checksum of the original archive remained the same, the new archive will be moved instead of the old one.
+In case the checksum is different an error will be returned: “Update failed. The configuration file was changed, try again later.“
```

Comment:
"Try again" should be enough, it is already "later". Also "The configuration changed" should probably be "The configuration was changed by somebody else."